src/games/seabattle/game.ts

```diff
@@ -244,6 +251,17 @@ export function getCellVector(a: ICell, b: ICell): ICell {
   return { x: a.x - b.x, y: a.y - b.y };
 }
 
+function validateShipsHaveUniqueIDs(ships: IShip[]): IShipsValidationResult {
+  const usedIDs: number[] = [];
+  for (const ship of ships) {
+    if (usedIDs.includes(ship.id)) {
```

Comment:
This is O(n^2) while you could easily do it using a hashmap instead of a list and get it to be O(n)